### PR TITLE
test: Add integration specs for views

### PR DESCRIPTION
**Pull request by gvgesanto2**
## Description

**Task:** https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md

This PR makes the following changes:
- Sets up [Capybara](http://teamcapybara.github.io/capybara/).
- Adds integration tests for the following views:
    - users index page.
    - user show page.
    - user posts index page.
    - user post show page.

## Type of change

- [X] Tests (changes related to testing)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have deleted any unused code as comments in my codebase

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   def index
```
Comment (FrederickMih) [inline app/controllers/posts_controller.rb:2]: - [ ] Hi @gvgesanto2, let's keep our application scalable by eliminating the N+1 query problems from our application(Seems n extra queries are being executed when fetching all posts and comments.) Kindly fix these issues.
- [ ] Kindly do the same for comments

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @gvgesanto2, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- All features tests for your pages are passing :heavy_check_mark: 
- Correct GitHub flow :+1: 
- PR has a title and summary :heavy_check_mark: 
- No Linter error :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gvgesanto2** [comment]
## Description

**Task:** https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md

This PR addresses the following change requested by the [last review](https://github.com/gvgesanto2/microverse-rails-blog/pull/8#pullrequestreview-1132073799):

> * [ ]  Hi @gvgesanto2, let's keep our application scalable by eliminating the N+1 query problems from our application(Seems n extra queries are being executed when fetching all posts and comments.) Kindly fix these issues.
> * [ ]  Kindly do the same for comments

**Solution adopted:** It was used the Bullet gem to detect all the N+1 query issues in the application and the ``includes()`` method to solve them.

## Type of change

- [X] Performance (non-breaking change which fixes performance issues)
- [ ] Test (change related to testing)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have deleted any unused code as comments in my codebase

**Review by DeliceLydia — APPROVED** [review]
## Hi @gvgesanto2, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**